src/Elasticsearch.Extensions.Logging/Options/ElasticsearchLoggerOptions.cs

```diff
@@ -68,5 +69,11 @@ public class ElasticsearchLoggerOptions : IEcsDocumentCreationOptions
 		/// 'Production', etc).
 		/// </summary>
 		public string[]? Tags { get; set; }
+
+		/// <summary>
+		/// Allows the direct setting of a <see cref="HttpTransport{TConfiguration}"/> to be used to communicate with Elasticsearch.
+		/// <para>If set takes precedence over <see cref="ShipTo"/> </para>
+		/// </summary>
+		public HttpTransport? Transport { get; set; }
```

Comment:
@Mpdreamz  please don't do this, as it breaks the Options pattern. All options need to be simple types, so that they can be set from configuration sections, i.e. JSON.

I know I haven't looked at this for a while, but configuration options really need to be kept configurable, i.e. be able to be set through the configuration system (files, commands line, environment variables, Azure web app deployments, KeyVault, etc).  This basically means strings (environment variables are strings), or simple types.

Note that the default extension method `loggingBuilder.AddElasticsearch()` automatically uses the configuration section "Logging:Elasticsearch", so you don't need any special C# code, and everything can be done via config.

In some dynamic situations that may not be possible, which is fine, but in those cases use a different extension method / constructor. And if at all possible, also try and express the same options via configuration (strings).

Maybe add a comment to the file about preserving the Options pattern and only use strings, or simple types that can be expressed as strings (arrays and dictionaries are also allowed; basically whatever fits in JSON).